### PR TITLE
Add Stage/Object Collidable setting

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

To support 3D visualization use cases in Habitat, this PR adds a flag and getter/setters to enable/disable collisions for objects and stages. This is done by manipulating the collision shape allowing all MotionType properties regardless of collidable setting. Flag exposed to Simulator API in C++ and Python as well as Object and Stage Attributes and config files.

Includes minor bugfix:
 - Friction/restitution/damping set on objects after construction were getting overwritten by initialization attributes when changing object MotionType.

## How Has This Been Tested

Unit tests added for config value parsing, contact and raycast tests w/ non-collidable objects and stage.

Local testing results in viewer:
- 3 stacks of objects with bottom object of different MotionType: dynamic(left), kinematic(middle), static(right).
- Procedure:
  - objects are simulated until sleeping
  - bottom of each stack is set to non-collidable (notice Bullet debug disappears and other objects fall through)
  - all objects are set to DYNAMIC (notice non-collidable objects fall through the ground)
  - gravity is inverted and non-collidable objects "fall" upward through the ground
  - gravity is returned to normal and all objects are set to collidable, falling into neat stacks.

collidable-test

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
Looks good to me assuming you get CI passing.

**Comment by Skylion007** [comment]
LGTM, let's merge. 👍

File: src/esp/sim/Simulator.h [hunk 1]
```diff
@@ -515,6 +515,46 @@ class Simulator {
    */
   bool contactTest(int objectID, int sceneID = 0);
 
+  /**
+   * @brief Set an object to collidable or not.
+   */
+  bool setObjectIsCollidable(bool collidable, const int objectID) {
+    if (sceneHasPhysics(0)) {
```
Comment (jturner65) [inline src/esp/sim/Simulator.h:522]: perhaps change 0 to "activeSceneID_"

**Review by jturner65 — APPROVED** [review]
One minor request.  Otherwise LGTM

**Comment by aclegg3** [comment]
Added unit tests, will merge once everything passes.

**Pull request merged**